backend/src/controllers/user.ts

```diff
@@ -25,23 +21,19 @@ export const login = async (req: express.Request, res: express.Response) => {
 			email,
 			password,
 		} = req.body
-	if(!email || !password) return res.sendStatus(400);
-	try {
-		const user = await db.lookupUserByEmail(email);
-		const expectedHash = authentication(user.salt, password);
-		if(expectedHash !== user.password) return res.sendStatus(403);
+	if(!verifyParameters(email, password)) return res.sendStatus(400);
+	const user = await db.lookupUserByEmail(email);
+	const expectedHash = authentication(user.salt, password);
+	if(expectedHash !== user.password) return res.sendStatus(403);
 
-		const newSalt = random();
-		user.salt = newSalt;
-		user.sessionToken = authentication(newSalt, user.username);
+	const newSalt = random();
+	user.salt = newSalt;
+	user.sessionToken = authentication(newSalt, user.username);
 
-		await db.save(user);
+	await db.save(user);
 
-		res.cookie('user-auth', user.sessionToken, { domain: "localhost", path: "/" });
-		return res.sendStatus(200);
-	} catch {
-		return res.sendStatus(404);
-	}
+	res.cookie('user-auth', user.sessionToken, { domain: "localhost", path: "/" });
+	return res.sendStatus(200);
```

Comment:
Not sure whether this try/catch is just for db failures or if the auth might fail too